src/module.c

```diff
@@ -10846,10 +10998,407 @@ sds genModulesInfoString(sds info) {
+}
+
+/* This is a series of parseAndSet functions for each type that, for the most part, emulate the ones
+ * in config.c. The difference here is that we use the set callback specified by the module
+ * to set the value */
+int setModuleBoolConfig(ModuleConfig *config, int val, const char **err) {
+    int return_code = config->set_fn.set_bool(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+int setModuleStringConfig(ModuleConfig *config, sds strval, const char **err) {
+    RedisModuleString *prev = config->get_fn.get_string(config->name, config->privdata);
+    RedisModuleString *new = createStringObject(strval, strlen(strval));
+    int return_code = config->set_fn.set_string(config->name, new, config->privdata, err);
+    if (prev) decrRefCount(prev);
+    decrRefCount(new);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+int setModuleEnumConfig(ModuleConfig *config, int val, const char **err) {
+    int return_code = config->set_fn.set_enum(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+int setModuleNumericConfig(ModuleConfig *config, long long val, const char **err) {
+    int return_code = config->set_fn.set_numeric(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+/* This is a series of getAndParse functions for each type that, for the most part, emulate the ones
+ * in config.c. The difference here is that we use the get callbacks specified by the module
+ * to get the value. */
+int getModuleBoolConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_bool(module_config->name, module_config->privdata);
+}
+
+sds getModuleStringConfig(ModuleConfig *module_config) {
+    RedisModuleString *val = module_config->get_fn.get_string(module_config->name, module_config->privdata);
+    if (val)
+        return sdsdup(val->ptr);
+    return NULL;
+}
+
+int getModuleEnumConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_enum(module_config->name, module_config->privdata);
+}
+
+long long getModuleNumericConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_numeric(module_config->name, module_config->privdata);
+}
+/*
+This function takes a module and a list of configs stored as sds NAME VALUE pairs.
+ * It attempts to call set on each of these configs. */
+int loadModuleConfigs(RedisModule *module) {
+    if (!server.module_configs_queue || !listLength(server.module_configs_queue)) return REDISMODULE_OK;
+    listIter li;
+    listNode *ln;
+    int count;
+    sds *config_pair;
+    const char *err = NULL;
+
+    listRewind(server.module_configs_queue, &li);
+    while ((ln = listNext(&li))) {
+        sds module_config = ln->value;
+        if (!(strcasestr(module_config, module->name) == module_config)) {
+            continue;
+        }
+        config_pair = sdssplitargs(module_config, &count);
```

Comment:
This is likely not binary data safe, since sdssplitargs will split on spaces. If the value has a space it will break, we probably need to store this as something other than an sds pair. (Like a tuple)